app/javascript/components/SweetInput/SweetInput.js

```diff
@@ -34,20 +36,26 @@ export default class SweetInput extends Component {
     errorMessage: '',
   };
 
-  onChange(value: string) {
-    if (this.props.onChange) {
-      this.props.onChange(value);
-    }
-  }
-
   hasError() {
+    const { validationState, errorMessage } = this.props;
+    if (validationState || validationState === null) {
+      return validationState === 'error';
+    }
     return !!this.props.errorMessage;
   }
```

Comment:
Hey @eyko, it's not clear to me how the API for SweetInput behaves given the `validationState` and `errorMessage` variables. Isn't `errorMessage` enough?

Also, I think you could simplify the hasError method by writing it like this:

```
return validationState === 'error' || validationState === null || !!this.props.ErrorMessage;
```

Reply:
I wanted to maintain backwards compatibility for now, so wherever we're using `errorMessage` still works as it used to, but it prints a deprecation warning in development. 

If we use the new property `validationState`, we can set it to one `success`, `warning`, `error`, or `null`. At the moment, I'm only checking for errors: `hasError()` only returns true if `validationState` is present and is `error`, or if it is _not_ present and we have an `errorMessage`. 

```js
return validationState === 'error' || validationState === null || ...
```

would return true if `validationState` is null (which we don't want!)

Reply:
I see! Let's try to remove the usage of errorMessage while QA is happening. Also, I noticed we're not using the different validationStates, wouldn't just a boolean flag be enough. What if we have a `hasError` prop instead of the `validationState`.

And re: the condition, you're right. 

We could still omit the null check tho:
```
return validationState === 'error' || !!this.props.ErrorMessage;
```

Reply:
You're right, I was being overly cautious with the `validationState`. It's unlikely that we would pass both `validationState` and `errorMessage` to the component (and it would be wrong to do so anyway) so the null check is unnecessary.

The reason for the other possible values was that I wanted to also add support for "green" success and "orange/yellowish" warning states, like with bootstrap / other css frameworks. I've just checked the codebase, however, and we don't seem to have any of that in the entire project so I think it's better to go for the boolean approach.

Reply:
:thumbsup: